pkg/controller/test-images-distributor/test_images_distributor_test.go

```diff
@@ -259,13 +271,14 @@ func TestReconcile(t *testing.T) {
 	outdatedImageStream := func() *imagev1.ImageStream {
 		copy := expectedImageStream.DeepCopy()
 		copy.Spec.LookupPolicy.Local = false
+		copy.ObjectMeta.Annotations["foo"] = "baz"
```

Comment:
@alvaroaleman any clue why this doesn't get updated?

```
=== FAIL: pkg/controller/test-images-distributor TestReconcile/Outdated_Imagestream_is_updated,_import_is_created (0.04s)
    --- FAIL: TestReconcile/Outdated_Imagestream_is_updated,_import_is_created (0.04s)
        test_images_distributor_test.go:609: verification failed: actual imagestream differs from expected:   &v1.ImageStream{
              	... // 1 ignored field
              	ObjectMeta: v1.ObjectMeta{
              		... // 1 ignored and 8 identical fields
              		DeletionGracePeriodSeconds: nil,
              		Labels:                     nil,
            - 		Annotations:                map[string]string{"foo": "bar"},
            + 		Annotations:                map[string]string{"foo": "baz"},
              		OwnerReferences:            nil,
              		Finalizers:                 nil,
              		... // 2 identical fields
              	},
              	Spec:   {LookupPolicy: {Local: true}},
              	Status: {},
              }
```

The upsert uses a deep-equality so I'm not clear why it wouldn't call Update

Reply:
Because you must set the Annotations in the `mutateFN`, not in the initial stream object. That is only used to identify the object (Line 305 -> L402).